coffeeshop-service/src/main/java/me/escoffier/quarkus/coffeeshop/CoffeeShopResource.java

```diff
@@ -24,7 +25,14 @@
 
     @PostMapping("/http")
     public Uni<Beverage> http(Order order) {
-        return barista.order(order.setOrderId(getId()));
+        return barista.order(order.setOrderId(getId()))
+                .onItem().invoke(beverage -> beverage.setPreparationState(Beverage.State.READY))
+                .ifNoItem().after(Duration.ofMillis(1500)).recoverWithItem(createFallbackBeverage(order))
```

Comment:
let's have some fun...

```java
.onItem().invoke(beverage -> beverage.setPreparationState(Beverage.State.READY))
                .ifNoItem().after(Duration.ofMillis(1500)).fail()
                .onFailure().recoverWithItem(createFallbackBeverage(order));
```

Reply:
does `.fail()` then trigger the `onFailure()`?

Reply:
Yes, that the beauty of it :-)

Reply:
That's pretty sweet